Fruit_Jam/Larsio_Paint_Music/sound_manager.py

```diff
@@ -96,20 +108,51 @@ def __init__(self, audio_output="pwm", seconds_per_eighth=0.25):
                     wsel_pin = board.I2S_WS
                     din_pin = board.I2S_DIN
 
+                # Check if DAC is connected
+                for i in range(500): # try for 5 seconds
+                    if i2c.try_lock():
+                        break
+                    time.sleep(0.01)
+                if 0x18 in i2c.scan():
+                    ltv320_present = True
+                else:
+                    ltv320_present = False
+                i2c.unlock()
+
                 # Initialize TLV320
-                self.tlv = adafruit_tlv320.TLV320DAC3100(i2c)
-                self.tlv.configure_clocks(sample_rate=11025, bit_depth=16)
-                self.tlv.headphone_output = True
-                self.tlv.headphone_volume = -15  # dB
-
-                # Setup I2S audio output - important to do this AFTER configuring the DAC
-                self.audio = audiobusio.I2SOut(
-                    bit_clock=bclck_pin,
-                    word_select=wsel_pin,
-                    data=din_pin
-                )
-
-                print("TLV320 I2S DAC initialized successfully")
+                if ltv320_present:
```

Comment:
I think it's fine to leave initialization of the TLV320 unguarded by the if statement and above logic checking for it. 

This project specifies Fruit Jam in the name and the TLV is built-in so safe to assume it's there. The original Metro version that shown in the guide also uses the TLV320 so I think it's fine for the code to go ahead and init without checking.

Since this project is sound based and documented with the TLV I think it's best to have it error out if the TLV is not found rather than try to fall back to a different audio type that isn't documented anywhere.

Reply:
makes sense, I'll hop on this later tonight.